server/src/dtos/asset-media-response.dto.ts

```diff
@@ -0,0 +1,42 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { IsOptional } from 'class-validator';
+
+export enum AssetMediaStatusEnum {
+  UPDATED = 'updated',
+  DUPLICATE = 'duplicate',
+}
+
+export class DefaultAssetMediaResponseDto {
+  @ApiProperty({
+    type: String,
+    enum: AssetMediaStatusEnum,
+    required: true,
+    enumName: 'AssetMediaStatus',
+  })
+  readonly status?: string;
+  @IsOptional()
+  assetId?: string;
+  @IsOptional()
+  copyId?: string;
+  @IsOptional()
+  duplicateId?: string;
+}
+
+export class AssetMediaUpdateResponseDto {
+  readonly status = 'updated';
+  assetId: string;
+  copyId: string;
+  constructor(assetId: string, copyId: string) {
+    this.assetId = assetId;
+    this.copyId = copyId;
+  }
+}
+export class DuplicateAssetResponseDto {
+  readonly status = 'duplicate';
+  duplicateId: string;
+  constructor(duplicateId: string) {
+    this.duplicateId = duplicateId;
+  }
+}
```

Comment:
Generally, we don't actually instantiate instances of these dtos, they act purely as types. The only reason they have to be classes is because they need to exist at runtime for the swagger plugin to work correctly. I would prefer to just send { status: 'duplicate', duplicateId: assetId } instead of creating an instance and having code for the constructor.